start-training

```diff
@@ -133,24 +137,33 @@ try:
         if args.wandb:
             import wandb
             if wandb.login():
+                entity = "safelife"
+                if args.project is None:
+                    project = args.env_type.replace("curriculum-", "")
+                elif "/" in args.project:
+                    entity, _slash, project = args.project.partition("/")
+                else:
+                    project = args.project
+
                 run_notes = os.path.join(safety_dir, 'run-notes.txt')
                 if os.path.exists(run_notes):
                     run_notes = open(run_notes).read()
                 else:
                     run_notes = None
-                wandb.init(reinit=True, name=subjob_name, config=args, notes=run_notes)
+
+                wandb.init(reinit=True, name=subjob_name, config=args,
+                           entity=entity, project=project, notes=run_notes)
                 wandb.config.penalty = penalty
 
         training_envs, testing_envs = build_environments(
-            args.env_type, args.run_type, main_seed, data_logger,
-            penalty, args.penalty_baseline)
+            args, main_seed, data_logger, penalty)
```

Comment:
I somewhat prefer to have the arguments explicitly laid out rather than lumping the entirety of `args` into a single argument. First, it makes it more obvious what parameters are actually being used to build the environment, so the code is a little easier to read. Second, it makes the code a little less fragile. If `args` is passed around to separate files, then if you change some naming convention in `args` you also have to make sure to change the name in the other file as well, and that can be easy/annoying to mess up.

Reply:
Hah, I feel the other way :) For a small number of parameters (<4?) it's okay to pass them in one by one, but at some point it starts to be a really crazy code pattern to have growing laundry lists of arguments at every function call in a stack, leading to unwieldy code and lots of extra editing when variables are either added or deleted. (In this change I was adding one more parameter, so I bundled them)

I view the name space of a config object as an API of sorts, and only change it when needed, though `git grep \\.old_parameter_name` works pretty well if you need to make such a change, even without a type system.

I think most of the community is [agrees with the bundled approach](https://softwareengineering.stackexchange.com/questions/145055/are-there-guidelines-on-how-many-parameters-a-function-should-accept)!

Reply:
Fair enough! Let's add the penalty to args then so that we're consistent.